samples/sensor/hts221/src/main.c

```diff
@@ -9,49 +9,105 @@
 #include <drivers/sensor.h>
 #include <stdio.h>
 #include <sys/util.h>
+#include <logging/log.h>
+LOG_MODULE_REGISTER(app);
 
 static void process_sample(struct device *dev)
 {
 	static unsigned int obs;
 	struct sensor_value temp, hum;
+
 	if (sensor_sample_fetch(dev) < 0) {
-		printf("Sensor sample update error\n");
+		LOG_ERR("Sensor sample update error");
 		return;
 	}
 
 	if (sensor_channel_get(dev, SENSOR_CHAN_AMBIENT_TEMP, &temp) < 0) {
-		printf("Cannot read HTS221 temperature channel\n");
+		LOG_ERR("Cannot read HTS221 temperature channel");
 		return;
 	}
 
 	if (sensor_channel_get(dev, SENSOR_CHAN_HUMIDITY, &hum) < 0) {
-		printf("Cannot read HTS221 humidity channel\n");
+		LOG_ERR("Cannot read HTS221 humidity channel");
 		return;
 	}
 
-	++obs;
-	printf("Observation:%u\n", obs);
+	obs++;
+	LOG_INF("Observation:%u", obs);
 
 	/* display temperature */
-	printf("Temperature:%.1f C\n", sensor_value_to_double(&temp));
+	LOG_INF("Temperature:%d.%08d C", temp.val1, temp.val2);
 
 	/* display humidity */
-	printf("Relative Humidity:%.1f%%\n",
-	       sensor_value_to_double(&hum));
+	LOG_INF("Relative Humidity:%d.%08d", hum.val1, hum.val2);
+}
+
+static void sensor_callback(struct device *dev, enum sensor_channel chan,
+				int res, void *user_data)
+{
+	process_sample(dev);
 }
 
+static void expired(struct k_timer *timer)
+{
+	int err = sensor_sample_fetch_async_chan(k_timer_user_data_get(timer),
+					SENSOR_CHAN_ALL, sensor_callback,
+					NULL);
+	if (err != 0) {
+		LOG_ERR("err:%d", err);
+	}
+}
+
+K_TIMER_DEFINE(timer, expired, NULL);
+
 static void hts221_handler(struct device *dev,
 			   struct sensor_trigger *trig)
 {
 	process_sample(dev);
 }
 
+#ifdef CONFIG_PPI_TRACE
```

Comment:
Any pointers (or a Kconfig fragment) for how to run with this enabled on nrf52_pca20020 (which is how I assume you are testing this)? I tried running this sample from this branch inside NCS with this config overlay, but I never seem to get to main():

```
CONFIG_USE_SEGGER_RTT=y

# CONFIG_SHELL_BACKEND_RTT is not set

CONFIG_LOG=y
# CONFIG_LOG_BACKEND_SHOW_COLOR is not set
CONFIG_LOG_BACKEND_RTT=y

# CONFIG_UART_CONSOLE is not set
CONFIG_RTT_CONSOLE=y


CONFIG_PPI_TRACE=y

```

Reply:
> Any pointers (or a Kconfig fragment) for how to run with this enabled on nrf52_pca20020 (which is how I assume you are testing this)?

@nordic-krch ping?

Reply:
@mbolivar sorry, got involved in something else. Will look at it tomorrow. Maybe I forgot to commit something.

Reply:
No problem, I will keep trying to figure it out for myself. Thanks.

Reply:
i tried to run it and on my setup it also does not work well. After enabling logging i see that k_timer setup in main.c is constantly expiring even though it is set to expire 1sec. Unfortunately, i ended up on sick leave. Will get back into that once i get better.

Reply:
OK, thanks, and I hope you get better soon!